docker-compose.yml

```diff
@@ -1,15 +1,22 @@
 services:
   postgres:
-    image: postgres:16
+    container_name: hostedgpt_postgres
+    build:
+      context: .
+      dockerfile: docker/Dockerfile.postgres
     restart: always
     environment:
-      POSTGRES_USER: app
-      POSTGRES_DB: app_development
-      POSTGRES_PASSWORD: secret
+      - POSTGRES_USER=${POSTGRES_USER:-app}
+      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-secret}
+      - POSTGRES_DB=${POSTGRES_DB:-app_development}
+      # NOTE: If DATABASE_URL is set it will override values set in HOSTEDGPT_DB_USERNAME and HOSTEDGPT_DB_PASSWORD
+      - DATABASE_URL
+      - HOSTEDGPT_DB_USERNAME=${HOSTEDGPT_DB_USERNAME:-app}
+      - HOSTEDGPT_DB_PASSWORD=${HOSTEDGPT_DB_PASSWORD:-secret}
     volumes:
       - hostedgpt_pgdata:/var/lib/postgresql/data
     healthcheck:
-      test: ["CMD", "pg_isready", "-U", "app", "-d", "app_development"]
+      test: pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB
```

Comment:
had to go back to using postgres_db because if the database already exists then the init process doesn't run and pg_isready was trying to then connect to the "app" database (matching the "app" user) which doesn't exist if you have an existing deployment. Existing deployments will have a db named app_development.